transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/TransportPlugin.java

```diff
@@ -47,18 +47,21 @@
 public class TransportPlugin implements Plugin<Project> {
 
   public void apply(Project project) {
+
+    TransportPluginConfig extension = project.getExtensions().create("transportPlugin", TransportPluginConfig.class, project);
```

Comment:
Rename the extension to `transport`? Most plugins I have seen don't have the word `plugin` as part of the extension name.

Reply:
Now that you mention it neither have I. Will do!

Reply:
To your comment here- Thanks for suggestion, will make that change.

As for the top-level comment:

Because the config parameters affect the configuration of individual tasks, and do not serve as input properties to the tasks themselves, I am not sure that we will ever be able to use extensions to configure these specific parameters. 

This is because when the plugin is applied in the build script, we immediately make use of the parameter values, as opposed to assigning them as properties to the tasks we are configuring. 

Typically I've seen extensions used to configure parameters that are passed into the tasks themselves, which may be useful to have later on, because the buildscript is able to alter their state before the task is actually executed. 

Interested to know your thoughts here.

Reply:
I am not sure if that is fully true, I have seen Gradle's code quality plugins like Checkstyle use the `sourceSets` property in it's extension to create tasks itself and not as inputs to its tasks. I am not sure if Checkstyle goes against Gradle's best practices for extension (I am assuming it doesn't). Our use case looks very similar to the Checkstyle use case in that regard.